doc/source/administrator/optimization.md

```diff
@@ -383,3 +383,141 @@ Choosing the right _limit to guarantee ratio_ ratio is an art, not a science.
+| prePuller.hook.resources                | hook-image-awaiter            | 0, 0                             | The container just polls the k8s api-server. Will require minimal resources.                                                                                                                                                                                                                                                           |
+| singleuser.cpu\|memory.guarantee\|limit | jupyter-username              | 0, 1G                            | The configuration syntax is different because it is native to the Spawner base class rather than Kubernetes. It is commonly useful to guarantee a certain amount of memory rather than CPU to help users share CPU with each other.                                                                                                    |
+
+### Example resource requests for high reliability
+
+Here are some examples of resource requests that could be reasonable for a
+deployment where service reliability and performance matters more than
+minimizing the required capacity on nodes.
+
+As these have been estimated roughly, please help us improve them by providing
+feedback in [this GitHub issue](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/2229).
+
+```{admonition} Note
+If you collect metrics on the resource usage of pods in your k8s cluster
+(Prometheus) and have dashboards showing you its usage (Grafana), you could
+tweak these over time.
+```
+
+```yaml
+hub: # hub pod, running jupyterhub/jupyterhub
+  resources:
+    requests:
+      cpu: 1
+      memory: 512Mi
+proxy:
+  chp: # proxy pod, running jupyterhub/configurable-http-proxy
+    resources:
+      requests:
+        cpu: 1
+        memory: 512Mi
+  traefik: # autohttps pod (optional, running traefik/traefik)
+    resources:
+      requests:
+        cpu: 100m
+        memory: 256Mi
+  secretSync: # autohttps pod (optional, sidecar container running small Python script)
+    resources:
+      requests:
+        cpu: 10m
+        memory: 64Mi
+scheduling:
+  userScheduler: # user-scheduler pods (optional, running kubernetes/kube-scheduler)
+    resources:
+      requests:
+        cpu: 20m
+        memory: 128Mi
+  userPlaceholder: # user-placeholder pods (optional, running pause container)
+    # This is just an override of the resource requests that otherwise match
+    # those configured in singleuser.cpu|memory.guarantee|limit.
+    resources: {}
+prePuller: # hook|continuous-image-puller pods (optional, running pause container)
+  resources:
+    requests:
+      cpu: 10m
+      memory: 8Mi
+  hook: # hook-image-awaiter pod (optional, running GoLang binary defined in images/image-awaiter)
+    resources:
+      requests:
+        cpu: 10m
+        memory: 8Mi
```

Comment:
@yuvipanda does these suggestions look reasonable to you for a deployment under heavy load / stress test?

Reply:
Looking at https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2034#issuecomment-782262510 I realize that the requests set here are too low to ensure they get what they need for great performance.

The amount of memory and cpu varied a bit more than I thought. I'm really not sure what to suggest here... Hmm....